lib/apiservers/engine/backends/filter/container.go

```diff
@@ -87,31 +84,59 @@ func IncludeContainer(listContext *ContainerListContext, container *models.Conta
 	}
 
 	// filter on network name
-	for n := range container.Endpoints {
-		name := container.Endpoints[n].Scope
-		if listContext.Filters.Include("network") {
-			err := listContext.Filters.WalkValues("network", func(value string) error {
-				if name == value {
-					return nil
-				}
-				return fmt.Errorf("not part of container network")
-			})
-			if err != nil {
-				return ExcludeAction
+	if listContext.Filters.Include("network") {
+		netFilterValues := listContext.Filters.Get("network")
+
+		// Gather the container's networks in a map
+		numNetworks := len(container.Endpoints)
```

Comment:
I observed this (interesting) behavior with Docker 1.13:

```
$ docker create -v foo:/dir busybox
6adc0f4ce8ced54692b839cd9a640371b0f46fdea4c95d3129dcb128f8656446
$ docker ps -a -f volume=foo
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
6adc0f4ce8ce        busybox             "sh"                16 seconds ago      Created                                 sleepy_clarke
$ docker ps -a -f volume=foo -f network=bar
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
$ docker ps -a -f network=bar -f volume=foo
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
$ docker ps -a -f volume=foo -f volume=bar
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
6adc0f4ce8ce        busybox             "sh"                About a minute ago   Created                                 sleepy_clarke
$ docker ps -a -f volume=foo -f volume=bar -f network=buz
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

It looks like `volume` and `network` filters are OR ops by themselves, but if used together they're not and their filter conditions must be satisfied individually.